Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -154,8 +154,10 @@ void VertexLoaderX64::ReadColor(OpArg data, u64 attribute, int format, int eleme
 		case FORMAT_32B_888x:
 		case FORMAT_32B_8888:
 			MOV(32, R(scratch1), data);
-			// See VertexLoader_Color.cpp for a comment on this condition.
-			if (format != FORMAT_32B_8888 || !elements)
+			// Resetting the alpha of 8888 when attribute != DIRECT breaks e.g.:
+			// - Mario Party 4 (which part?)
+			// - the opening menu of Xenoblade Chronicles
+			if (format != FORMAT_32B_8888 || (attribute == DIRECT && !elements))
```

Comment:
Do we know this is actually correct vs. just "if (format != FORMAT_32B_8888)"?  If it is correct as-is, are all the other  color formats handling this bit correctly?  It doesn't make any sense to have both FORMAT_32B_888x and a separate bit which controls exactly the same thing.